pavelib/utils/test/suites/pytest_suite.py

```diff
@@ -174,31 +158,44 @@ def _default_test_id(self):
         using a default test id.
         """
         # We need to use $DIR/*, rather than just $DIR so that
-        # django-nose will import them early in the test process,
+        # pytest will import them early in the test process,
         # thereby making sure that we load any django models that are
         # only defined in test files.
-        default_test_id = (
-            "{system}/djangoapps/*"
-            " common/djangoapps/*"
-            " openedx/core/djangoapps/*"
-            " openedx/tests/*"
-            " openedx/core/lib/*"
-        )
-
+        default_test_globs = [
+            "{system}/djangoapps/*".format(system=self.root),
+            "common/djangoapps/*",
+            "openedx/core/djangoapps/*",
+            "openedx/tests/*",
+            "openedx/core/lib/*",
+        ]
         if self.root in ('lms', 'cms'):
-            default_test_id += " {system}/lib/*"
+            default_test_globs.append("{system}/lib/*".format(system=self.root))
 
         if self.root == 'lms':
-            default_test_id += " {system}/tests.py"
-            default_test_id += " openedx/core/djangolib/*"
-            default_test_id += " openedx/features"
-
-        return default_test_id.format(system=self.root)
-
-
-class LibTestSuite(NoseTestSuite):
+            default_test_globs.append("{system}/tests.py".format(system=self.root))
+            default_test_globs.append("openedx/core/djangolib/*")
+            default_test_globs.append("openedx/features")
+
+        def included(path):
+            """
+            Should this path be included in the pytest arguments?
+            """
+            if path.endswith('__pycache__'):
+                return False
+            return path.endswith('.py') or os.path.isdir(path)
+
+        default_test_paths = []
+        for path_glob in default_test_globs:
+            if '*' in path_glob:
+                default_test_paths += [path for path in glob(path_glob) if included(path)]
+            else:
+                default_test_paths += [path_glob]
+        return ' '.join(default_test_paths)
```

Comment:
One of the biggest changes I make in my subsequent tox changes is to dispense with this globbing logic and just pass the top-level directories to pytest for its discovery. I've so far left the non-tox pytest runs the same, using this globbing logic. Do you see problems with using the top-level directories for tox?

Reply:
If I remember correctly, the main reason I switched it is that it was that having the shell expand the globs included some directories and files which pytest would return error codes if run on, causing the overall job to fail.  I was reluctant to just pass in the specific subdirectories under `openedx/code/djangolib` because that could miss new subdirectories added there if the list isn't updated.  Ideally it should just work if we run all the tests under `openedx/core/djangolib` in one test run, but I thought it could take a while to debug existing bugs in setup/teardown running them that way.